diff --git a/bin/runLightSalome.sh b/bin/runLightSalome.sh
index 4caad1db..af913073 100755
--- a/bin/runLightSalome.sh
+++ b/bin/runLightSalome.sh
@@ -242,7 +242,7 @@ run_light_salome()
     export LD_LIBRARY_PATH
     if [ "$(uname -s)" = "Darwin" ]; then
         DYLD_LIBRARY_PATH=${my_ld_library_path}:${DYLD_LIBRARY_PATH}
-        DYLD_LIBRARY_PATH=$(echo ${DYLD_LIBRARY_PATH} | sed -i "" -e "s,^:,,;s,:$,,;s,::\+,:,g")
+        DYLD_LIBRARY_PATH=$(echo ${DYLD_LIBRARY_PATH} | sed -e "s,^:,,;s,:$,,;s,::\+,:,g")
         export DYLD_LIBRARY_PATH
     fi
     PYTHONPATH=${my_pythonpath}:${PYTHONPATH}
diff --git a/src/Qtx/Qtx.h b/src/Qtx/Qtx.h
index 6917bd2c..166a0329 100644
--- a/src/Qtx/Qtx.h
+++ b/src/Qtx/Qtx.h
@@ -288,7 +288,7 @@ public:
 
   static QString     getenv( const QString& envVar );
 
-#ifndef WIN32 && !defined(__APPLE__)
+#if !defined(WIN32) && !defined(__APPLE__)
   static void*       getDisplay();
   static Qt::HANDLE  getVisual();
 #endif
