diff --git a/src/NETGENPlugin/NETGENPlugin_Hypothesis.cxx b/src/NETGENPlugin/NETGENPlugin_Hypothesis.cxx
index 509b15d..1a4b94f 100644
--- a/src/NETGENPlugin/NETGENPlugin_Hypothesis.cxx
+++ b/src/NETGENPlugin/NETGENPlugin_Hypothesis.cxx
@@ -26,6 +26,11 @@
 // Date      : 28/03/2006
 // Project   : SALOME
 //
+
+#if defined(__APPLE__) && defined(__arm64__)
+#include <mach/arm/thread_status.h>
+#endif
+
 #include "NETGENPlugin_Hypothesis.hxx"
 
 #include "NETGENPlugin_Mesher.hxx"
diff --git a/src/NETGENPlugin/NETGENPlugin_Mesher.cxx b/src/NETGENPlugin/NETGENPlugin_Mesher.cxx
index d003888..d477719 100644
--- a/src/NETGENPlugin/NETGENPlugin_Mesher.cxx
+++ b/src/NETGENPlugin/NETGENPlugin_Mesher.cxx
@@ -27,6 +27,10 @@
 // Project   : SALOME
 //=============================================================================
 
+#if defined(__APPLE__) && defined(__arm64__)
+#include <mach/arm/thread_status.h>
+#endif
+
 #include "NETGENPlugin_Mesher.hxx"
 #include "NETGENPlugin_Hypothesis_2D.hxx"
 #include "NETGENPlugin_SimpleHypothesis_3D.hxx"
diff --git a/src/NETGENPlugin/NETGENPlugin_Mesher.hxx b/src/NETGENPlugin/NETGENPlugin_Mesher.hxx
index dc8d77b..05644b5 100644
--- a/src/NETGENPlugin/NETGENPlugin_Mesher.hxx
+++ b/src/NETGENPlugin/NETGENPlugin_Mesher.hxx
@@ -75,7 +75,10 @@ namespace netgen {
 class ChdirRAII
 {
   public:
-#ifndef WIN32
+#ifdef __APPLE__
+    ChdirRAII(const std::string& wd):_wd(wd) { if(_wd.empty()) return ; char *pwd = getcwd(NULL, 0); _od = pwd; free(pwd); chdir(_wd.c_str()); }
+    ~ChdirRAII() { if(_od.empty()) return ; chdir(_od.c_str()); }
+#elifndef WIN32
     ChdirRAII(const std::string& wd):_wd(wd) { if(_wd.empty()) return ; char *pwd(get_current_dir_name()); _od = pwd; free(pwd); chdir(_wd.c_str()); }
     ~ChdirRAII() { if(_od.empty()) return ; chdir(_od.c_str()); }
 #else
diff --git a/src/NETGENPlugin/NETGENPlugin_NETGEN_2D3D.cxx b/src/NETGENPlugin/NETGENPlugin_NETGEN_2D3D.cxx
index d01805b..139264e 100644
--- a/src/NETGENPlugin/NETGENPlugin_NETGEN_2D3D.cxx
+++ b/src/NETGENPlugin/NETGENPlugin_NETGEN_2D3D.cxx
@@ -27,6 +27,11 @@
 // Project   : SALOME
 //=============================================================================
 //
+
+#if defined(__APPLE__) && defined(__arm64__)
+#include <mach/arm/thread_status.h>
+#endif
+
 #include "NETGENPlugin_NETGEN_2D3D.hxx"
 #include "NETGENPlugin_Hypothesis.hxx"
 #include "NETGENPlugin_SimpleHypothesis_3D.hxx"
diff --git a/src/NETGENPlugin/NETGENPlugin_NETGEN_3D.cxx b/src/NETGENPlugin/NETGENPlugin_NETGEN_3D.cxx
index e0d033d..4cca2df 100644
--- a/src/NETGENPlugin/NETGENPlugin_NETGEN_3D.cxx
+++ b/src/NETGENPlugin/NETGENPlugin_NETGEN_3D.cxx
@@ -28,6 +28,11 @@
 // Project   : SALOME
 //=============================================================================
 //
+
+#if defined(__APPLE__) && defined(__arm64__)
+#include <mach/arm/thread_status.h>
+#endif
+
 #include "NETGENPlugin_NETGEN_3D.hxx"
 
 #include "NETGENPlugin_Hypothesis.hxx"
diff --git a/src/NETGENPlugin/NETGENPlugin_NETGEN_3D_Remote.cxx b/src/NETGENPlugin/NETGENPlugin_NETGEN_3D_Remote.cxx
index 7b3bd9b..beca095 100644
--- a/src/NETGENPlugin/NETGENPlugin_NETGEN_3D_Remote.cxx
+++ b/src/NETGENPlugin/NETGENPlugin_NETGEN_3D_Remote.cxx
@@ -28,6 +28,10 @@
 //=============================================================================
 //
 //
+#if defined(__APPLE__) && defined(__arm64__)
+#include <mach/arm/thread_status.h>
+#endif
+
 #include "NETGENPlugin_NETGEN_3D_Remote.hxx"
 
 #include "NETGENPlugin_NETGEN_3D.hxx"
diff --git a/src/NETGENPlugin/NETGENPlugin_NETGEN_3D_SA.cxx b/src/NETGENPlugin/NETGENPlugin_NETGEN_3D_SA.cxx
index 7f6df71..0c4aa2d 100644
--- a/src/NETGENPlugin/NETGENPlugin_NETGEN_3D_SA.cxx
+++ b/src/NETGENPlugin/NETGENPlugin_NETGEN_3D_SA.cxx
@@ -28,6 +28,11 @@
 //=============================================================================
 //
 //
+
+#if defined(__APPLE__) && defined(__arm64__)
+#include <mach/arm/thread_status.h>
+#endif
+
 #include "NETGENPlugin_NETGEN_3D_SA.hxx"
 
 #include "NETGENPlugin_DriverParam.hxx"
diff --git a/src/NETGENPlugin/NETGENPlugin_i.cxx b/src/NETGENPlugin/NETGENPlugin_i.cxx
index 9fa3705..a5f246b 100644
--- a/src/NETGENPlugin/NETGENPlugin_i.cxx
+++ b/src/NETGENPlugin/NETGENPlugin_i.cxx
@@ -26,6 +26,11 @@
 //  Module : SMESH
 //  $Header$
 //
+
+#if defined(__APPLE__) && defined(__arm64__)
+#include <mach/arm/thread_status.h>
+#endif
+
 #include "utilities.h"
 
 #include "NETGENPlugin_Hypothesis_2D_ONLY_i.hxx"
