--- qt-5.15.19_old/qtbase/mkspecs/common/mac.conf	2026-09-16 09:38:18
+++ qt-5.15.19_new/qtbase/mkspecs/common/mac.conf	2026-09-16 09:47:02
@@ -18,8 +18,7 @@ QMAKE_INCDIR_OPENGL     = \
 
 # sdk.prf will prefix the proper SDK sysroot
 QMAKE_INCDIR_OPENGL     = \
-    /System/Library/Frameworks/OpenGL.framework/Headers \
-    /System/Library/Frameworks/AGL.framework/Headers/
+    /System/Library/Frameworks/OpenGL.framework/Headers
 
 QMAKE_FIX_RPATH         = install_name_tool -id
 
@@ -30,7 +29,7 @@ QMAKE_LIBS_DYNLOAD      =
 QMAKE_REL_RPATH_BASE    = @loader_path
 
 QMAKE_LIBS_DYNLOAD      =
-QMAKE_LIBS_OPENGL       = -framework OpenGL -framework AGL
+QMAKE_LIBS_OPENGL       = -framework OpenGL
 QMAKE_LIBS_THREAD       =
 
 QMAKE_INCDIR_WAYLAND    =
